<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
    <meta charset="utf-8">
    <meta name="generator" content="quarto-1.5.31">

    <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

    <meta name="author" content="Piero Palacios Bernuy">
    <meta name="dcterms.date" content="2024-04-30">
    <meta name="keywords" content="Genomic Ranges, Omics, Bioconductor">

    <title>Integrative Analysis of Multi-omic Data</title>
    <style>
      code{white-space: pre-wrap;}
      span.smallcaps{font-variant: small-caps;}
      div.columns{display: flex; gap: min(4vw, 1.5em);}
      div.column{flex: auto; overflow-x: auto;}
      div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
      ul.task-list{list-style: none;}
      ul.task-list li input[type="checkbox"] {
        width: 0.8em;
        margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
        vertical-align: middle;
      }
      /* CSS for syntax highlighting */
      pre > code.sourceCode { white-space: pre; position: relative; }
      pre > code.sourceCode > span { line-height: 1.25; }
      pre > code.sourceCode > span:empty { height: 1.2em; }
      .sourceCode { overflow: visible; }
      code.sourceCode > span { color: inherit; text-decoration: inherit; }
      div.sourceCode { margin: 1em 0; }
      pre.sourceCode { margin: 0; }
      @media screen {
      div.sourceCode { overflow: auto; }
      }
      @media print {
      pre > code.sourceCode { white-space: pre-wrap; }
      pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
      }
      pre.numberSource code
        { counter-reset: source-line 0; }
      pre.numberSource code > span
        { position: relative; left: -4em; counter-increment: source-line; }
      pre.numberSource code > span > a:first-child::before
        { content: counter(source-line);
          position: relative; left: -1em; text-align: right; vertical-align: baseline;
          border: none; display: inline-block;
          -webkit-touch-callout: none; -webkit-user-select: none;
          -khtml-user-select: none; -moz-user-select: none;
          -ms-user-select: none; user-select: none;
          padding: 0 4px; width: 4em;
        }
      pre.numberSource { margin-left: 3em;  padding-left: 4px; }
      div.sourceCode
        {   }
      @media screen {
      pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
      }
      /* CSS for citations */
      div.csl-bib-body { }
      div.csl-entry {
        clear: both;
      }
      .hanging-indent div.csl-entry {
        margin-left:2em;
        text-indent:-2em;
      }
      div.csl-left-margin {
        min-width:2em;
        float:left;
      }
      div.csl-right-inline {
        margin-left:2em;
        padding-left:1em;
      }
      div.csl-indent {
        margin-left: 2em;
      }    </style>

    <style>
      body.hypothesis-enabled #quarto-embed-header {
        padding-right: 36px;
      }

      #quarto-embed-header {
        height: 3em;
        width: 100%;
        display: flex;
        justify-content: space-between;
        align-items: center;
        border-bottom: solid 1px;
      }

      #quarto-embed-header h6 {
        font-size: 1.1em;
        padding-top: 0.6em;
        margin-left: 1em;
        margin-right: 1em;
        font-weight: 400;
      }

      #quarto-embed-header a.quarto-back-link,
      #quarto-embed-header a.quarto-download-embed {
        font-size: 0.8em;
        margin-top: 1em;
        margin-bottom: 1em;
        margin-left: 1em;
        margin-right: 1em;
      }

      .quarto-back-container {
        padding-left: 0.5em;
        display: flex;
      }

      .headroom {
          will-change: transform;
          transition: transform 200ms linear;
      }

      .headroom--pinned {
          transform: translateY(0%);
      }

      .headroom--unpinned {
          transform: translateY(-100%);
      }      
    </style>

    <script>
    window.document.addEventListener("DOMContentLoaded", function () {

      var header = window.document.querySelector("#quarto-embed-header");
      const titleBannerEl = window.document.querySelector("body > #title-block-header");
      if (titleBannerEl) {
        titleBannerEl.style.paddingTop = header.clientHeight + "px";
      }
      const contentEl = window.document.getElementById('quarto-content');
      for (const child of contentEl.children) {
        child.style.paddingTop = header.clientHeight + "px";
        child.style.marginTop = "1em";
      }

      // Use the article root if the `back` call doesn't work. This isn't perfect
      // but should typically work
      window.quartoBackToArticle = () => {
        var currentUrl = window.location.href;
        window.history.back();
        setTimeout(() => {
            // if location was not changed in 100 ms, then there is no history back
            if(currentUrl === window.location.href){              
                // redirect to site root
                window.location.href = "index.html";
            }
        }, 100);
      }

      const headroom = new window.Headroom(header, {
        tolerance: 5,
        onPin: function () {
        },
        onUnpin: function () {
        },
      });
      headroom.init();
    });
    </script>

    
<script src="site_libs/manuscript-notebook/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script src="site_libs/quarto-contrib/glightbox/glightbox.min.js"></script>
<link href="site_libs/quarto-contrib/glightbox/glightbox.min.css" rel="stylesheet">
<link href="site_libs/quarto-contrib/glightbox/lightbox.css" rel="stylesheet">
     <script async="" src="https://hypothes.is/embed.js"></script>
<script>
  window.document.addEventListener("DOMContentLoaded", function (_event) {
    document.body.classList.add('hypothesis-enabled');
  });
</script>  <script src="site_libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<script src="site_libs/plotly-binding-4.10.4/plotly.js"></script>
<script src="site_libs/typedarray-0.1/typedarray.min.js"></script>
<script src="site_libs/jquery-3.5.1/jquery.min.js"></script>
<link href="site_libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet">
<script src="site_libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>
<link href="site_libs/plotly-htmlwidgets-css-2.11.1/plotly-htmlwidgets.css" rel="stylesheet">
<script src="site_libs/plotly-main-2.11.1/plotly-latest.min.js"></script>
<link href="site_libs/htmltools-fill-0.5.7/fill.css" rel="stylesheet">
<script src="site_libs/bslib-component-js-0.6.1/components.min.js"></script>
<script src="site_libs/bslib-component-js-0.6.1/web-components.min.js" type="module"></script>
<link href="site_libs/bslib-component-css-0.6.1/components.css" rel="stylesheet">
<script src="site_libs/bslib-tag-require-0.6.1/tag-require.js"></script>  
      <meta name="citation_title" content="Integrative Analysis of Multi-omic Data">
<meta name="citation_abstract" content="This document is part of a series of the analysis of Omics data. Especifically, here is showed how to analyze bulk RNA-Seq data with Bioconductor packages. Also, it&amp;amp;#039;s showcased how to make plots of the RNA data in the context of differentially gene expression and gene-sets.
">
<meta name="citation_keywords" content="Genomic Ranges,Omics,Bioconductor">
<meta name="citation_author" content="Piero Palacios Bernuy">
<meta name="citation_publication_date" content="2024-04-30">
<meta name="citation_cover_date" content="2024-04-30">
<meta name="citation_year" content="2024">
<meta name="citation_online_date" content="2024-04-30">
<meta name="citation_language" content="en">
<meta name="citation_journal_title" content="An open source portfolio">
<meta name="citation_reference" content="citation_title=Strategies for the development of volcanic hazard maps in monogenetic volcanic fields: The example of La Palma (Canary Islands);,citation_author=José Marrero;,citation_author=Alicia García;,citation_author=Manuel Berrocoso;,citation_author=Ángeles Llinares;,citation_author=Antonio Rodríguez-Losada;,citation_author=R. Ortiz;,citation_publication_date=2019-07;,citation_cover_date=2019-07;,citation_year=2019;,citation_doi=10.1186/s13617-019-0085-5;,citation_volume=8;,citation_journal_title=Journal of Applied Volcanology;">
</head>

  <body class="quarto-notebook">
    <div id="quarto-embed-header" class="headroom fixed-top bg-primary">
      
      <a onclick="window.quartoBackToArticle(); return false;" class="btn btn-primary quarto-back-link" href=""><i class="bi bi-caret-left"></i> Back to Article</a>
      <h6><i class="bi bi-journal-code"></i> Article Notebook</h6>

            <a href="./index.qmd" class="btn btn-primary quarto-download-embed" download="index.qmd">Download Source</a>
          </div>

     <header id="title-block-header" class="quarto-title-block default toc-left page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">Integrative Analysis of Multi-omic Data</h1>
          </div>

    
    <div class="quarto-title-meta-container">
      <div class="quarto-title-meta-column-start">
        
        <div class="quarto-title-meta">

                <div>
            <div class="quarto-title-meta-heading">Author</div>
            <div class="quarto-title-meta-contents">
                        <p>Piero Palacios Bernuy <a href="mailto:p.palacios.bernuy@gmail.com" class="quarto-title-author-email"><i class="bi bi-envelope"></i></a> <a href="https://orcid.org/0000-0001-6729-4080" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a></p>
                      </div>
          </div>
                
                <div>
            <div class="quarto-title-meta-heading">Published</div>
            <div class="quarto-title-meta-contents">
              <p class="date">April 30, 2024</p>
            </div>
          </div>
          
                
              </div>
      </div>
      <div class="quarto-title-meta-column-end quarto-other-formats-target">
      </div>
    </div>

    <div>
      <div class="abstract">
        <div class="block-title">Abstract</div>
        <p>This document is part of a series of the analysis of Omics data. Especifically, here is showed how to analyze bulk RNA-Seq data with Bioconductor packages. Also, it’s showcased how to make plots of the RNA data in the context of differentially gene expression and gene-sets.</p>
      </div>
    </div>

    <div>
      <div class="keywords">
        <div class="block-title">Keywords</div>
        <p>Genomic Ranges, Omics, Bioconductor</p>
      </div>
    </div>

    <div class="quarto-other-links-text-target">
    </div>  </div>
</header><div id="quarto-content" class="page-columns page-rows-contents page-layout-article toc-left">
<div id="quarto-sidebar-toc-left" class="sidebar toc-left">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#introduction" id="toc-introduction" class="nav-link active" data-scroll-target="#introduction"><span class="header-section-number">1</span> Introduction</a></li>
  <li><a href="#gwas-catalog-with-a-chip-seq-experiment" id="toc-gwas-catalog-with-a-chip-seq-experiment" class="nav-link" data-scroll-target="#gwas-catalog-with-a-chip-seq-experiment"><span class="header-section-number">2</span> GWAS Catalog with a ChIP-Seq Experiment</a></li>
  <li><a href="#explore-the-tcga" id="toc-explore-the-tcga" class="nav-link" data-scroll-target="#explore-the-tcga"><span class="header-section-number">3</span> Explore the TCGA</a></li>
  <li><a href="#conclusion" id="toc-conclusion" class="nav-link" data-scroll-target="#conclusion"><span class="header-section-number">5</span> Conclusion</a></li>
  <li><a href="#references" id="toc-references" class="nav-link" data-scroll-target="#references">References</a></li>
  </ul>
</nav>
</div>
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar zindex-bottom">
</div>
<main class="content quarto-banner-title-block" id="quarto-document-content">      

       <section id="introduction" class="level2" data-number="1">
<h2 data-number="1" class="anchored" data-anchor-id="introduction"><span class="header-section-number">1</span> Introduction</h2>
</section>
<section id="gwas-catalog-with-a-chip-seq-experiment" class="level2" data-number="2">
<h2 data-number="2" class="anchored" data-anchor-id="gwas-catalog-with-a-chip-seq-experiment"><span class="header-section-number">2</span> GWAS Catalog with a ChIP-Seq Experiment</h2>
<p>Here, we are gonna analyze the relation between transcription factor binding (ESRRA binding data) from a ChIP-Seq experiment and the genome-wide associations between DNA variants and phenotypes like diseases. For this task, we are gonna use a the <code>gwascat</code> package distributed by the <strong>EMBL</strong> (European Molecular Biology Laboratories).</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [1]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'ggplot2' was built under R version 4.3.2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'tidyr' was built under R version 4.3.2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'readr' was built under R version 4.3.2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'purrr' was built under R version 4.3.2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'dplyr' was built under R version 4.3.2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'stringr' was built under R version 4.3.2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'lubridate' was built under R version 4.3.2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.0     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
</div>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(gwascat)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'gwascat' was built under R version 4.3.1</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>gwascat loaded.  Use makeCurrentGwascat() to extract current image.
 from EBI.  The data folder of this package has some legacy extracts.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GenomeInfoDb)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'GenomeInfoDb' was built under R version 4.3.2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: BiocGenerics</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'BiocGenerics' was built under R version 4.3.1</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'BiocGenerics'

The following objects are masked from 'package:lubridate':

    intersect, setdiff, union

The following objects are masked from 'package:dplyr':

    combine, intersect, setdiff, union

The following objects are masked from 'package:stats':

    IQR, mad, sd, var, xtabs

The following objects are masked from 'package:base':

    anyDuplicated, aperm, append, as.data.frame, basename, cbind,
    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,
    get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,
    match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,
    Position, rank, rbind, Reduce, rownames, sapply, setdiff, sort,
    table, tapply, union, unique, unsplit, which.max, which.min

Loading required package: S4Vectors
Loading required package: stats4

Attaching package: 'S4Vectors'

The following objects are masked from 'package:lubridate':

    second, second&lt;-

The following objects are masked from 'package:dplyr':

    first, rename

The following object is masked from 'package:tidyr':

    expand

The following object is masked from 'package:utils':

    findMatches

The following objects are masked from 'package:base':

    expand.grid, I, unname

Loading required package: IRanges

Attaching package: 'IRanges'

The following object is masked from 'package:lubridate':

    %within%

The following objects are masked from 'package:dplyr':

    collapse, desc, slice

The following object is masked from 'package:purrr':

    reduce

The following object is masked from 'package:grDevices':

    windows</code></pre>
</div>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ERBS)</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(liftOver)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: GenomicRanges
Loading required package: rtracklayer</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'rtracklayer' was built under R version 4.3.1</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: Homo.sapiens
Loading required package: AnnotationDbi</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'AnnotationDbi' was built under R version 4.3.2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: Biobase
Welcome to Bioconductor

    Vignettes contain introductory material; view with
    'browseVignettes()'. To cite Bioconductor, see
    'citation("Biobase")', and for packages 'citation("pkgname")'.


Attaching package: 'AnnotationDbi'

The following object is masked from 'package:dplyr':

    select

Loading required package: OrganismDbi</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'OrganismDbi' was built under R version 4.3.1</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: GenomicFeatures</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'GenomicFeatures' was built under R version 4.3.2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: GO.db

Loading required package: org.Hs.eg.db

Loading required package: TxDb.Hsapiens.UCSC.hg19.knownGene</code></pre>
</div>
</div></div>
<p>First, we need to download the data, keep the 24 chromosomes (from 1 to Y) and, specify the sequence information from the GRCh38 human genome annotation.</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [2]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>gwcat <span class="ot">=</span> <span class="fu">get_cached_gwascat</span>()</span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a>gg <span class="ot">=</span> gwcat <span class="sc">|&gt;</span> <span class="fu">as_GRanges</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>dropping 45505 records that have NA for CHR_POS</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>1950 records have semicolon in CHR_POS; splitting and using first entry.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>3265 records have ' x ' in CHR_POS indicating multiple SNP effects, using first.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a>gg <span class="ot">=</span> <span class="fu">keepStandardChromosomes</span>(gg, <span class="at">pruning.mode =</span> <span class="st">"coarse"</span>)</span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a><span class="co"># seqlevelsStyle(gg) &lt;- "UCSC"</span></span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a><span class="fu">seqlevels</span>(gg) <span class="ot">&lt;-</span> <span class="fu">seqlevels</span>(gg) <span class="sc">|&gt;</span> </span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sortSeqlevels</span>()</span>
<span id="cb32-7"><a href="#cb32-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-8"><a href="#cb32-8" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">"si.hs.38"</span>)</span>
<span id="cb32-9"><a href="#cb32-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-10"><a href="#cb32-10" aria-hidden="true" tabindex="-1"></a><span class="fu">seqinfo</span>(gg) <span class="ot">&lt;-</span> si.hs<span class="fl">.38</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<p>Now, let’s plot a karyogram that will show the SNP’s identified with significant associations with a phenotype. The SNP’s in the GWAS catalog have a stringent criterion of significance and there has been a replication of the finding from a independent population.</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [3]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>ggbio<span class="sc">::</span><span class="fu">autoplot</span>(gg, <span class="at">layout=</span><span class="st">"karyogram"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Registered S3 method overwritten by 'GGally':
  method from   
  +.gg   ggplot2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Scale for x is already present.
Adding another scale for x, which will replace the existing scale.
Scale for x is already present.
Adding another scale for x, which will replace the existing scale.</code></pre>
</div>
<div class="cell-output-display">
<div class="">
<figure class="figure">
<p class=""><a href="index_files/figure-html/unnamed-chunk-3-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-1"><img src="index_files/figure-html/unnamed-chunk-3-1.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
</div></div>
<p>We can see the peak data as a <code>GRanges</code> object:</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [4]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">"GM12878"</span>)</span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a>GM12878</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>GRanges object with 1873 ranges and 7 metadata columns:
         seqnames            ranges strand |      name     score       col
            &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;integer&gt; &lt;logical&gt;
     [1]     chrX   1509355-1512462      * |         5         0      &lt;NA&gt;
     [2]     chrX 26801422-26802448      * |         6         0      &lt;NA&gt;
     [3]    chr19 11694102-11695359      * |         1         0      &lt;NA&gt;
     [4]    chr19   4076893-4079276      * |         4         0      &lt;NA&gt;
     [5]     chr3 53288568-53290767      * |         9         0      &lt;NA&gt;
     ...      ...               ...    ... .       ...       ...       ...
  [1869]    chr19 11201120-11203985      * |      8701         0      &lt;NA&gt;
  [1870]    chr19   2234920-2237370      * |       990         0      &lt;NA&gt;
  [1871]     chr1 94311336-94313543      * |      4035         0      &lt;NA&gt;
  [1872]    chr19 45690614-45691210      * |     10688         0      &lt;NA&gt;
  [1873]    chr19   6110100-6111252      * |      2274         0      &lt;NA&gt;
         signalValue    pValue    qValue      peak
           &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;integer&gt;
     [1]      157.92   310.000        32      1991
     [2]      147.38   310.000        32       387
     [3]       99.71   311.660        32       861
     [4]       84.74   310.000        32      1508
     [5]       78.20   299.505        32      1772
     ...         ...       ...       ...       ...
  [1869]        8.65     7.281   0.26576      2496
  [1870]        8.65    26.258   1.99568      1478
  [1871]        8.65    12.511   1.47237      1848
  [1872]        8.65     6.205   0.00000       298
  [1873]        8.65    17.356   2.01323       496
  -------
  seqinfo: 93 sequences (1 circular) from hg19 genome</code></pre>
</div>
</div></div>
<p>If we see the bottom of the <code>GRanges</code> table, this experiment have the hg19 annotation from the human genome. To work on the GRCh38 annotation we need to lift-over with a <code>.chain</code> file. For this we can use the <code>AnnotationHub</code> package.</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [5]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(AnnotationHub)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'AnnotationHub' was built under R version 4.3.1</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: BiocFileCache</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'BiocFileCache' was built under R version 4.3.1</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: dbplyr</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'dbplyr' was built under R version 4.3.2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'dbplyr'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:dplyr':

    ident, sql</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'AnnotationHub'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:Biobase':

    cache</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:rtracklayer':

    hubUrl</code></pre>
</div>
<div class="sourceCode cell-code" id="cb49"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a>ah <span class="ot">&lt;-</span> AnnotationHub<span class="sc">::</span><span class="fu">AnnotationHub</span>()</span>
<span id="cb49-2"><a href="#cb49-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-3"><a href="#cb49-3" aria-hidden="true" tabindex="-1"></a><span class="fu">query</span>(ah, <span class="fu">c</span>(<span class="st">"hg19ToHg38.over.chain"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>AnnotationHub with 1 record
# snapshotDate(): 2023-10-23
# names(): AH14150
# $dataprovider: UCSC
# $species: Homo sapiens
# $rdataclass: ChainFile
# $rdatadateadded: 2014-12-15
# $title: hg19ToHg38.over.chain.gz
# $description: UCSC liftOver chain file from hg19 to hg38
# $taxonomyid: 9606
# $genome: hg19
# $sourcetype: Chain
# $sourceurl: http://hgdownload.cse.ucsc.edu/goldenpath/hg19/liftOver/hg19To...
# $sourcesize: NA
# $tags: c("liftOver", "chain", "UCSC", "genome", "homology") 
# retrieve record with 'object[["AH14150"]]' </code></pre>
</div>
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a>chain <span class="ot">&lt;-</span> ah[[<span class="st">"AH14150"</span>]]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>loading from cache</code></pre>
</div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [6]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a>GM12878 <span class="ot">&lt;-</span> <span class="fu">liftOver</span>(GM12878, chain) <span class="sc">|&gt;</span> </span>
<span id="cb53-2"><a href="#cb53-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unlist</span>()</span>
<span id="cb53-3"><a href="#cb53-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-4"><a href="#cb53-4" aria-hidden="true" tabindex="-1"></a><span class="fu">seqlevelsStyle</span>(GM12878) <span class="ot">&lt;-</span> <span class="st">"NCBI"</span></span>
<span id="cb53-5"><a href="#cb53-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-6"><a href="#cb53-6" aria-hidden="true" tabindex="-1"></a><span class="fu">seqinfo</span>(GM12878) <span class="ot">&lt;-</span> si.hs<span class="fl">.38</span></span>
<span id="cb53-7"><a href="#cb53-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-8"><a href="#cb53-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-9"><a href="#cb53-9" aria-hidden="true" tabindex="-1"></a><span class="fu">seqlevelsStyle</span>(GM12878) <span class="ot">&lt;-</span> <span class="st">"UCSC"</span></span>
<span id="cb53-10"><a href="#cb53-10" aria-hidden="true" tabindex="-1"></a><span class="fu">seqlevelsStyle</span>(gg) <span class="ot">&lt;-</span> <span class="st">"UCSC"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<p>We can find overlaps between the GWAS catalog and the ESRRA ChIP-Seq experiment but, there is a problem; the GWAS catalog is a collection of intervals that reports all significant SNPs and there can be duplications of SNPs associated to multiple phenotypes or the same SNP might be found for the same phenotype in different studies.</p>
<p>We can see the duplications with the <code>reduce</code> function from <code>IRanges</code> package:</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [7]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb54"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a><span class="co"># duplicated loci</span></span>
<span id="cb54-2"><a href="#cb54-2" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(gg) <span class="sc">-</span> <span class="fu">length</span>(<span class="fu">reduce</span>(gg))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 261160</code></pre>
</div>
</div></div>
<p>We can see that there are <code>261160</code> duplicated loci. Let’s find the overlap between the <em>reduced</em> catalog and the ChIP-Seq experiment:</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [8]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb56-2"><a href="#cb56-2" aria-hidden="true" tabindex="-1"></a>fo <span class="ot">=</span> <span class="fu">findOverlaps</span>(GM12878, <span class="fu">reduce</span>(gg))</span>
<span id="cb56-3"><a href="#cb56-3" aria-hidden="true" tabindex="-1"></a>fo</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Hits object with 613 hits and 0 metadata columns:
        queryHits subjectHits
        &lt;integer&gt;   &lt;integer&gt;
    [1]         6      237757
    [2]         9      221130
    [3]        12       25060
    [4]        15      104699
    [5]        15      104700
    ...       ...         ...
  [609]      1928      246305
  [610]      1929      244698
  [611]      1929      244699
  [612]      1931      250380
  [613]      1931      250381
  -------
  queryLength: 1932 / subjectLength: 268560</code></pre>
</div>
</div></div>
<p>We can see 613 hits. Then, we are gonna eobtain the ranges from those hits, retrieve the phenotypes (DISEASE/TRAIT) and show the top 20 most common phenotypes with association to SNPs that lies on the ESRRA binding peaks.</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [9]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a>over_ranges <span class="ot">&lt;-</span> <span class="fu">reduce</span>(gg)[<span class="fu">subjectHits</span>(fo)]</span>
<span id="cb58-2"><a href="#cb58-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-3"><a href="#cb58-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-4"><a href="#cb58-4" aria-hidden="true" tabindex="-1"></a>ii <span class="ot">&lt;-</span> over_ranges <span class="sc">|&gt;</span> </span>
<span id="cb58-5"><a href="#cb58-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">|&gt;</span> </span>
<span id="cb58-6"><a href="#cb58-6" aria-hidden="true" tabindex="-1"></a>  GenomicRanges<span class="sc">::</span><span class="fu">makeGRangesListFromDataFrame</span>()</span>
<span id="cb58-7"><a href="#cb58-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-8"><a href="#cb58-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-9"><a href="#cb58-9" aria-hidden="true" tabindex="-1"></a>phset <span class="ot">=</span> <span class="fu">lapply</span>(ii, <span class="cf">function</span>(x){</span>
<span id="cb58-10"><a href="#cb58-10" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb58-11"><a href="#cb58-11" aria-hidden="true" tabindex="-1"></a>  <span class="co"># print(glue::glue("On range: {x}"))</span></span>
<span id="cb58-12"><a href="#cb58-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unique</span>(gg[<span class="fu">which</span>(gg <span class="sc">%over%</span> x)]<span class="sc">$</span><span class="st">"DISEASE/TRAIT"</span>)</span>
<span id="cb58-13"><a href="#cb58-13" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb58-14"><a href="#cb58-14" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb58-15"><a href="#cb58-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-16"><a href="#cb58-16" aria-hidden="true" tabindex="-1"></a>gwas_on_peaks <span class="ot">&lt;-</span> phset <span class="sc">|&gt;</span> </span>
<span id="cb58-17"><a href="#cb58-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">enframe</span>() <span class="sc">|&gt;</span> </span>
<span id="cb58-18"><a href="#cb58-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">unnest</span>(value) <span class="sc">|&gt;</span> </span>
<span id="cb58-19"><a href="#cb58-19" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">count</span>(value) <span class="sc">|&gt;</span> </span>
<span id="cb58-20"><a href="#cb58-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">slice_max</span>(n, <span class="at">n=</span><span class="dv">20</span>)</span>
<span id="cb58-21"><a href="#cb58-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-22"><a href="#cb58-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-23"><a href="#cb58-23" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> gwas_on_peaks <span class="sc">|&gt;</span> </span>
<span id="cb58-24"><a href="#cb58-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">value =</span> <span class="fu">fct_reorder</span>(value,n)) <span class="sc">|&gt;</span> </span>
<span id="cb58-25"><a href="#cb58-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(n,value, <span class="at">fill=</span>value))<span class="sc">+</span></span>
<span id="cb58-26"><a href="#cb58-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>() <span class="sc">+</span></span>
<span id="cb58-27"><a href="#cb58-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb58-28"><a href="#cb58-28" aria-hidden="true" tabindex="-1"></a>  paletteer<span class="sc">::</span><span class="fu">scale_fill_paletteer_d</span>(<span class="st">"khroma::smoothrainbow"</span>) <span class="sc">+</span></span>
<span id="cb58-29"><a href="#cb58-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span>
<span id="cb58-30"><a href="#cb58-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-31"><a href="#cb58-31" aria-hidden="true" tabindex="-1"></a>bslib<span class="sc">::</span><span class="fu">card</span>(plotly<span class="sc">::</span><span class="fu">ggplotly</span>(p), <span class="at">full_screen =</span> T)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="card bslib-card bslib-mb-spacing html-fill-item html-fill-container" data-bslib-card-init="" data-full-screen="false" data-require-bs-caller="card()" data-require-bs-version="5">
<div class="card-body bslib-gap-spacing html-fill-item html-fill-container" style="margin-top:auto;margin-bottom:auto;flex:1 1 auto;">
<div class="plotly html-widget html-fill-item" id="htmlwidget-171c168db4a533b58e5b" style="width:100%;height:400px;"></div>
<script type="application/json" data-for="htmlwidget-171c168db4a533b58e5b">{"x":{"data":[{"orientation":"v","width":7,"base":0.55000000000000004,"x":[3.5],"y":[0.89999999999999991],"text":"n:  7<br />value: Appendicular lean mass<br />value: Appendicular lean mass","type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(232,236,251,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"Appendicular lean mass","legendgroup":"Appendicular lean mass","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":7,"base":1.55,"x":[3.5],"y":[0.90000000000000013],"text":"n:  7<br />value: Diastolic blood pressure<br />value: Diastolic blood pressure","type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(221,216,239,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"Diastolic blood pressure","legendgroup":"Diastolic blood pressure","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":7,"base":2.5499999999999998,"x":[3.5],"y":[0.90000000000000036],"text":"n:  7<br />value: Educational attainment<br />value: Educational attainment","type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(209,193,225,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"Educational attainment","legendgroup":"Educational attainment","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":7,"base":3.5499999999999998,"x":[3.5],"y":[0.90000000000000036],"text":"n:  7<br />value: Lymphocyte count<br />value: Lymphocyte count","type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(195,168,209,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"Lymphocyte count","legendgroup":"Lymphocyte count","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":7,"base":4.5499999999999998,"x":[3.5],"y":[0.90000000000000036],"text":"n:  7<br />value: Mean spheric corpuscular volume<br />value: Mean spheric corpuscular volume","type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(181,143,194,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"Mean spheric corpuscular volume","legendgroup":"Mean spheric corpuscular volume","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":7,"base":5.5499999999999998,"x":[3.5],"y":[0.90000000000000036],"text":"n:  7<br />value: Monocyte count<br />value: Monocyte count","type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(167,120,180,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"Monocyte count","legendgroup":"Monocyte count","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":7,"base":6.5499999999999998,"x":[3.5],"y":[0.90000000000000036],"text":"n:  7<br />value: Plateletcrit<br />value: Plateletcrit","type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(155,98,167,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"Plateletcrit","legendgroup":"Plateletcrit","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":7,"base":7.5499999999999998,"x":[3.5],"y":[0.89999999999999947],"text":"n:  7<br />value: Reticulocyte fraction of red cells<br />value: Reticulocyte fraction of red cells","type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(140,78,153,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"Reticulocyte fraction of red cells","legendgroup":"Reticulocyte fraction of red cells","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":7,"base":8.5500000000000007,"x":[3.5],"y":[0.89999999999999858],"text":"n:  7<br />value: Rheumatoid arthritis<br />value: Rheumatoid arthritis","type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(111,76,155,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"Rheumatoid arthritis","legendgroup":"Rheumatoid arthritis","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":7,"base":9.5500000000000007,"x":[3.5],"y":[0.89999999999999858],"text":"n:  7<br />value: Triglyceride levels<br />value: Triglyceride levels","type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(96,89,169,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"Triglyceride levels","legendgroup":"Triglyceride levels","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":7,"base":10.550000000000001,"x":[3.5],"y":[0.89999999999999858],"text":"n:  7<br />value: Type 2 diabetes<br />value: Type 2 diabetes","type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(85,104,184,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"Type 2 diabetes","legendgroup":"Type 2 diabetes","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":8,"base":11.550000000000001,"x":[4],"y":[0.89999999999999858],"text":"n:  8<br />value: Heel bone mineral density<br />value: Heel bone mineral density","type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(78,121,197,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"Heel bone mineral density","legendgroup":"Heel bone mineral density","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":8,"base":12.550000000000001,"x":[4],"y":[0.89999999999999858],"text":"n:  8<br />value: Systolic blood pressure<br />value: Systolic blood pressure","type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(77,138,198,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"Systolic blood pressure","legendgroup":"Systolic blood pressure","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":9,"base":13.550000000000001,"x":[4.5],"y":[0.89999999999999858],"text":"n:  9<br />value: Sex hormone-binding globulin levels<br />value: Sex hormone-binding globulin levels","type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(78,150,188,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"Sex hormone-binding globulin levels","legendgroup":"Sex hormone-binding globulin levels","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":10,"base":14.550000000000001,"x":[5],"y":[0.89999999999999858],"text":"n: 10<br />value: Eosinophil counts<br />value: Eosinophil counts","type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(84,158,179,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"Eosinophil counts","legendgroup":"Eosinophil counts","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":10,"base":15.550000000000001,"x":[5],"y":[0.89999999999999858],"text":"n: 10<br />value: HDL cholesterol levels<br />value: HDL cholesterol levels","type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(89,165,169,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"HDL cholesterol levels","legendgroup":"HDL cholesterol levels","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":10,"base":16.550000000000001,"x":[5],"y":[0.89999999999999858],"text":"n: 10<br />value: Hematocrit<br />value: Hematocrit","type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(96,171,158,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"Hematocrit","legendgroup":"Hematocrit","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":10,"base":17.550000000000001,"x":[5],"y":[0.89999999999999858],"text":"n: 10<br />value: Platelet count<br />value: Platelet count","type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(105,177,144,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"Platelet count","legendgroup":"Platelet count","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":10,"base":18.550000000000001,"x":[5],"y":[0.89999999999999858],"text":"n: 10<br />value: Total cholesterol levels<br />value: Total cholesterol levels","type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(119,183,125,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"Total cholesterol levels","legendgroup":"Total cholesterol levels","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":10,"base":19.550000000000001,"x":[5],"y":[0.89999999999999858],"text":"n: 10<br />value: Waist-to-hip ratio adjusted for BMI<br />value: Waist-to-hip ratio adjusted for BMI","type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(140,188,104,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"Waist-to-hip ratio adjusted for BMI","legendgroup":"Waist-to-hip ratio adjusted for BMI","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":10,"base":20.550000000000001,"x":[5],"y":[0.89999999999999858],"text":"n: 10<br />value: White blood cell count<br />value: White blood cell count","type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(166,190,84,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"White blood cell count","legendgroup":"White blood cell count","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":11,"base":21.550000000000001,"x":[5.5],"y":[0.89999999999999858],"text":"n: 11<br />value: High density lipoprotein cholesterol levels<br />value: High density lipoprotein cholesterol levels","type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(190,188,72,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"High density lipoprotein cholesterol levels","legendgroup":"High density lipoprotein cholesterol levels","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":13,"base":22.550000000000001,"x":[6.5],"y":[0.89999999999999858],"text":"n: 13<br />value: Multi-trait sex score<br />value: Multi-trait sex score","type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(209,181,65,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"Multi-trait sex score","legendgroup":"Multi-trait sex score","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":14,"base":23.550000000000001,"x":[7],"y":[0.89999999999999858],"text":"n: 14<br />value: Body mass index<br />value: Body mass index","type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(221,170,60,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"Body mass index","legendgroup":"Body mass index","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":15,"base":24.550000000000001,"x":[7.5],"y":[0.89999999999999858],"text":"n: 15<br />value: Red cell distribution width<br />value: Red cell distribution width","type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(228,156,57,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"Red cell distribution width","legendgroup":"Red cell distribution width","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":16,"base":25.550000000000001,"x":[8],"y":[0.89999999999999858],"text":"n: 16<br />value: Mean corpuscular volume<br />value: Mean corpuscular volume","type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(231,140,53,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"Mean corpuscular volume","legendgroup":"Mean corpuscular volume","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":18,"base":26.550000000000001,"x":[9],"y":[0.89999999999999858],"text":"n: 18<br />value: Red blood cell count<br />value: Red blood cell count","type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(230,121,50,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"Red blood cell count","legendgroup":"Red blood cell count","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":25,"base":27.550000000000001,"x":[12.5],"y":[0.89999999999999858],"text":"n: 25<br />value: Mean corpuscular hemoglobin<br />value: Mean corpuscular hemoglobin","type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(228,99,45,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"Mean corpuscular hemoglobin","legendgroup":"Mean corpuscular hemoglobin","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":48,"base":28.550000000000001,"x":[24],"y":[0.89999999999999858],"text":"n: 48<br />value: Height<br />value: Height","type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(223,72,40,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"Height","legendgroup":"Height","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null}],"layout":{"margin":{"t":26.228310502283104,"r":7.3059360730593621,"b":40.182648401826491,"l":276.89497716894988},"font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724},"xaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[-2.4000000000000004,50.399999999999999],"tickmode":"array","ticktext":["0","10","20","30","40","50"],"tickvals":[0,10,20,30,40,50],"categoryorder":"array","categoryarray":["0","10","20","30","40","50"],"nticks":null,"ticks":"","tickcolor":null,"ticklen":3.6529680365296811,"tickwidth":0,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.68949771689498},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":true,"gridcolor":"rgba(235,235,235,1)","gridwidth":0.66417600664176002,"zeroline":false,"anchor":"y","title":{"text":"n","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724}},"hoverformat":".2f"},"yaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[0.40000000000000002,29.600000000000001],"tickmode":"array","ticktext":["Appendicular lean mass","Diastolic blood pressure","Educational attainment","Lymphocyte count","Mean spheric corpuscular volume","Monocyte count","Plateletcrit","Reticulocyte fraction of red cells","Rheumatoid arthritis","Triglyceride levels","Type 2 diabetes","Heel bone mineral density","Systolic blood pressure","Sex hormone-binding globulin levels","Eosinophil counts","HDL cholesterol levels","Hematocrit","Platelet count","Total cholesterol levels","Waist-to-hip ratio adjusted for BMI","White blood cell count","High density lipoprotein cholesterol levels","Multi-trait sex score","Body mass index","Red cell distribution width","Mean corpuscular volume","Red blood cell count","Mean corpuscular hemoglobin","Height"],"tickvals":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29],"categoryorder":"array","categoryarray":["Appendicular lean mass","Diastolic blood pressure","Educational attainment","Lymphocyte count","Mean spheric corpuscular volume","Monocyte count","Plateletcrit","Reticulocyte fraction of red cells","Rheumatoid arthritis","Triglyceride levels","Type 2 diabetes","Heel bone mineral density","Systolic blood pressure","Sex hormone-binding globulin levels","Eosinophil counts","HDL cholesterol levels","Hematocrit","Platelet count","Total cholesterol levels","Waist-to-hip ratio adjusted for BMI","White blood cell count","High density lipoprotein cholesterol levels","Multi-trait sex score","Body mass index","Red cell distribution width","Mean corpuscular volume","Red blood cell count","Mean corpuscular hemoglobin","Height"],"nticks":null,"ticks":"","tickcolor":null,"ticklen":3.6529680365296811,"tickwidth":0,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.68949771689498},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":true,"gridcolor":"rgba(235,235,235,1)","gridwidth":0.66417600664176002,"zeroline":false,"anchor":"x","title":{"text":"value","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724}},"hoverformat":".2f"},"shapes":[{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":1}],"showlegend":true,"legend":{"bgcolor":null,"bordercolor":null,"borderwidth":0,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.68949771689498},"title":{"text":"value","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724}}},"hovermode":"closest","barmode":"relative"},"config":{"doubleClick":"reset","modeBarButtonsToAdd":["hoverclosest","hovercompare"],"showSendToCloud":false},"source":"A","attrs":{"4a54687b65e1":{"x":{},"y":{},"fill":{},"type":"bar"}},"cur_data":"4a54687b65e1","visdat":{"4a54687b65e1":["function (y) ","x"]},"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.20000000000000001,"selected":{"opacity":1},"debounce":0},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
</div>
<bslib-tooltip placement="auto" bsoptions="[]" data-require-bs-version="5" data-require-bs-caller="tooltip()">
<template>Expand</template>
<span class="bslib-full-screen-enter badge rounded-pill"><svg xmlns="http://www.w3.org/2000/svg" viewbox="0 0 24 24" style="height:1em;width:1em;fill:currentColor;" aria-hidden="true" role="img"><path d="M20 5C20 4.4 19.6 4 19 4H13C12.4 4 12 3.6 12 3C12 2.4 12.4 2 13 2H21C21.6 2 22 2.4 22 3V11C22 11.6 21.6 12 21 12C20.4 12 20 11.6 20 11V5ZM4 19C4 19.6 4.4 20 5 20H11C11.6 20 12 20.4 12 21C12 21.6 11.6 22 11 22H3C2.4 22 2 21.6 2 21V13C2 12.4 2.4 12 3 12C3.6 12 4 12.4 4 13V19Z"></path></svg></span>
</bslib-tooltip>
<script data-bslib-card-init="">bslib.Card.initializeAllCards();</script>
</div>
</div>
</div></div>
<p>Distinct phenotypes identified on the peaks:</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [10]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb59"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(phset)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 613</code></pre>
</div>
</div></div>
<p>Now, how to do the inference of these phenotype on peaks of these b cells? We can use permutation on the genomic positions to test if the number of phenotypes found is due to chance or not.</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [11]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb61"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb61-1"><a href="#cb61-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ph525x)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: png</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: grid</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: VariantAnnotation</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'VariantAnnotation' was built under R version 4.3.2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: MatrixGenerics</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'MatrixGenerics' was built under R version 4.3.1</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: matrixStats</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'matrixStats' was built under R version 4.3.2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'matrixStats'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:Biobase':

    anyMissing, rowMedians</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:dplyr':

    count</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'MatrixGenerics'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:matrixStats':

    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,
    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,
    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,
    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,
    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,
    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,
    colWeightedMeans, colWeightedMedians, colWeightedSds,
    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,
    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,
    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,
    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,
    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,
    rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,
    rowWeightedMads, rowWeightedMeans, rowWeightedMedians,
    rowWeightedSds, rowWeightedVars</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:Biobase':

    rowMedians</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: SummarizedExperiment</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'SummarizedExperiment' was built under R version 4.3.1</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: Rsamtools</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'Rsamtools' was built under R version 4.3.1</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: Biostrings</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'Biostrings' was built under R version 4.3.2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: XVector</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'XVector' was built under R version 4.3.1</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'XVector'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:purrr':

    compact</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'Biostrings'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:grid':

    pattern</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:base':

    strsplit</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'VariantAnnotation'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:stringr':

    fixed</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:base':

    tabulate</code></pre>
</div>
<div class="sourceCode cell-code" id="cb92"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb92-1"><a href="#cb92-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(progress)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'progress' was built under R version 4.3.3</code></pre>
</div>
<div class="sourceCode cell-code" id="cb94"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb94-1"><a href="#cb94-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb94-2"><a href="#cb94-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb94-3"><a href="#cb94-3" aria-hidden="true" tabindex="-1"></a>n_iter <span class="ot">=</span> <span class="dv">100</span></span>
<span id="cb94-4"><a href="#cb94-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb94-5"><a href="#cb94-5" aria-hidden="true" tabindex="-1"></a>pb <span class="ot">&lt;-</span> progress_bar<span class="sc">$</span><span class="fu">new</span>(<span class="at">format =</span> <span class="st">"(:spin) [:bar] :percent [Elapsed time: :elapsedfull || Estimated time remaining: :eta]"</span>,</span>
<span id="cb94-6"><a href="#cb94-6" aria-hidden="true" tabindex="-1"></a>                       <span class="at">total =</span> n_iter,</span>
<span id="cb94-7"><a href="#cb94-7" aria-hidden="true" tabindex="-1"></a>                       <span class="at">complete =</span> <span class="st">"="</span>,   <span class="co"># Completion bar character</span></span>
<span id="cb94-8"><a href="#cb94-8" aria-hidden="true" tabindex="-1"></a>                       <span class="at">incomplete =</span> <span class="st">" "</span>, <span class="co"># Incomplete bar character</span></span>
<span id="cb94-9"><a href="#cb94-9" aria-hidden="true" tabindex="-1"></a>                       <span class="at">current =</span> <span class="st">"&gt;"</span>,    <span class="co"># Current bar character</span></span>
<span id="cb94-10"><a href="#cb94-10" aria-hidden="true" tabindex="-1"></a>                       <span class="at">clear =</span> <span class="cn">FALSE</span>,    <span class="co"># If TRUE, clears the bar when finish</span></span>
<span id="cb94-11"><a href="#cb94-11" aria-hidden="true" tabindex="-1"></a>                       <span class="at">width =</span> <span class="dv">100</span>)   </span>
<span id="cb94-12"><a href="#cb94-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb94-13"><a href="#cb94-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb94-14"><a href="#cb94-14" aria-hidden="true" tabindex="-1"></a>rsc <span class="ot">=</span> <span class="fu">sapply</span>(<span class="dv">1</span><span class="sc">:</span>n_iter, <span class="cf">function</span>(x) {</span>
<span id="cb94-15"><a href="#cb94-15" aria-hidden="true" tabindex="-1"></a>    pb<span class="sc">$</span><span class="fu">tick</span>()</span>
<span id="cb94-16"><a href="#cb94-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">length</span>(<span class="fu">findOverlaps</span>(<span class="fu">reposition</span>(GM12878), <span class="fu">reduce</span>(gg))) <span class="sc">|&gt;</span> </span>
<span id="cb94-17"><a href="#cb94-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">suppressWarnings</span>()</span>
<span id="cb94-18"><a href="#cb94-18" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb94-19"><a href="#cb94-19" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb94-20"><a href="#cb94-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb94-21"><a href="#cb94-21" aria-hidden="true" tabindex="-1"></a><span class="co"># compute prop with more overlaps than in observed data</span></span>
<span id="cb94-22"><a href="#cb94-22" aria-hidden="true" tabindex="-1"></a><span class="fu">mean</span>(rsc <span class="sc">&gt;</span> <span class="fu">length</span>(fo))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0</code></pre>
</div>
</div></div>
</section>
<section id="explore-the-tcga" class="level2" data-number="3">
<h2 data-number="3" class="anchored" data-anchor-id="explore-the-tcga"><span class="header-section-number">3</span> Explore the TCGA</h2>
<p>Please check the script below the image to see how to explore and get insights from the TCGA (The Cancer Genome Atlas) database.</p>
<div class="quarto-embed-nb-cell" data-notebook="/home/runner/work/Multi-omics/Multi-omics/notebooks/TCGA.qmd" data-notebook-title="Integrative Analysis with TCGA Data" data-notebook-cellid="cell-read-data">
<div class="cell-container"><div class="cell-decorator"><pre>In [12]:</pre></div><div id="read-data" class="cell" data-message="false">
<div class="sourceCode cell-code" id="cb96"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb96-1"><a href="#cb96-1" aria-hidden="true" tabindex="-1"></a>readData <span class="op">=</span> curatedTCGAData(<span class="st">"READ"</span>, </span>
<span id="cb96-2"><a href="#cb96-2" aria-hidden="true" tabindex="-1"></a>                           c(<span class="st">"RNASeq2GeneNorm"</span>, <span class="st">"Mutation"</span>, <span class="st">"Methylation_methyl450"</span>), </span>
<span id="cb96-3"><a href="#cb96-3" aria-hidden="true" tabindex="-1"></a>                           dry.run <span class="op">=</span> FALSE, version <span class="op">=</span> <span class="st">"2.1.1"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Working on: READ_Mutation-20160128see ?curatedTCGAData and browseVignettes('curatedTCGAData') for documentationloading from cacherequire("RaggedExperiment")Warning: package 'RaggedExperiment' was built under R version 4.3.1Working on: READ_RNASeq2GeneNorm-20160128see ?curatedTCGAData and browseVignettes('curatedTCGAData') for documentationloading from cacheWorking on: READ_Methylation_methyl450-20160128see ?curatedTCGAData and browseVignettes('curatedTCGAData') for documentationloading from cacherequire("rhdf5")Warning: package 'rhdf5' was built under R version 4.3.2see ?curatedTCGAData and browseVignettes('curatedTCGAData') for documentationloading from cacheLoading required package: HDF5ArrayWarning: package 'HDF5Array' was built under R version 4.3.2Loading required package: DelayedArrayWarning: package 'DelayedArray' was built under R version 4.3.1Loading required package: MatrixWarning: package 'Matrix' was built under R version 4.3.2
Attaching package: 'Matrix'The following object is masked from 'package:S4Vectors':

    expandThe following objects are masked from 'package:tidyr':

    expand, pack, unpackLoading required package: S4ArraysWarning: package 'S4Arrays' was built under R version 4.3.2Loading required package: abind
Attaching package: 'S4Arrays'The following object is masked from 'package:abind':

    abindThe following object is masked from 'package:base':

    rowsumLoading required package: SparseArrayWarning: package 'SparseArray' was built under R version 4.3.2
Attaching package: 'DelayedArray'The following object is masked from 'package:purrr':

    simplifyThe following objects are masked from 'package:base':

    apply, scale, sweep
Attaching package: 'HDF5Array'The following object is masked from 'package:rhdf5':

    h5lsWorking on: READ_colData-20160128see ?curatedTCGAData and browseVignettes('curatedTCGAData') for documentationloading from cacheWorking on: READ_sampleMap-20160128see ?curatedTCGAData and browseVignettes('curatedTCGAData') for documentationloading from cacheWorking on: READ_metadata-20160128see ?curatedTCGAData and browseVignettes('curatedTCGAData') for documentationloading from cacheharmonizing input:
  removing 1903 sampleMap rows not in names(experiments)
  removing 2 colData rownames not in sampleMap 'primary'</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>A MultiAssayExperiment object of 3 listed
 experiments with user-defined names and respective classes.
 Containing an ExperimentList class object of length 3:
 [1] READ_Mutation-20160128: RaggedExperiment with 22075 rows and 69 columns
 [2] READ_RNASeq2GeneNorm-20160128: SummarizedExperiment with 18115 rows and 177 columns
 [3] READ_Methylation_methyl450-20160128: SummarizedExperiment with 485577 rows and 106 columns
Functionality:
 experiments() - obtain the ExperimentList instance
 colData() - the primary/phenotype DataFrame
 sampleMap() - the sample coordination DataFrame
 `---
title: Integrative Analysis of Multi-omic Data
author:
  - name: Piero Palacios Bernuy
    orcid: 0000-0001-6729-4080
    corresponding: true
    email: p.palacios.bernuy@gmail.com
    roles:
      - Investigation
      - Bioinformatics
      - Deep learning
      - Visualization
keywords:
  - Genomic Ranges
  - Omics
  - Bioconductor
abstract: |
  This document is part of a series of the analysis of Omics data. Especifically, here is showed how to analyze bulk RNA-Seq data with Bioconductor packages. Also, it's showcased how to make plots of the RNA data in the context of differentially gene expression and gene-sets. 
plain-language-summary: |
  This document have a example of the analysis of bulk RNA-Seq data.
key-points:
  - A guide to analyze GWAS public data.
  - A guide to analyze TCGA database.
date: last-modified
bibliography: references.bib
citation:
  container-title: An open source portfolio
number-sections: true
---



## Introduction



## GWAS Catalog with a ChIP-Seq Experiment

Here,  we are gonna analyze the relation between transcription factor binding (ESRRA binding data) from a ChIP-Seq experiment and the genome-wide associations between DNA variants and phenotypes like diseases. For this task, we are gonna use a the `gwascat` package distributed by the **EMBL** (European Molecular Biology Laboratories).



::: {.cell}

```{.r .cell-code .hidden}
library(tidyverse)</code></pre>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'ggplot2' was built under R version 4.3.2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'tidyr' was built under R version 4.3.2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'readr' was built under R version 4.3.2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'purrr' was built under R version 4.3.2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'dplyr' was built under R version 4.3.2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'stringr' was built under R version 4.3.2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'lubridate' was built under R version 4.3.2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.0     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
</div>
<div class="sourceCode cell-code" id="cb107"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb107-1"><a href="#cb107-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(gwascat)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'gwascat' was built under R version 4.3.1</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>gwascat loaded.  Use makeCurrentGwascat() to extract current image.
 from EBI.  The data folder of this package has some legacy extracts.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb110"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb110-1"><a href="#cb110-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GenomeInfoDb)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'GenomeInfoDb' was built under R version 4.3.2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: BiocGenerics</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'BiocGenerics' was built under R version 4.3.1</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'BiocGenerics'

The following objects are masked from 'package:lubridate':

    intersect, setdiff, union

The following objects are masked from 'package:dplyr':

    combine, intersect, setdiff, union

The following objects are masked from 'package:stats':

    IQR, mad, sd, var, xtabs

The following objects are masked from 'package:base':

    anyDuplicated, aperm, append, as.data.frame, basename, cbind,
    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,
    get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,
    match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,
    Position, rank, rbind, Reduce, rownames, sapply, setdiff, sort,
    table, tapply, union, unique, unsplit, which.max, which.min

Loading required package: S4Vectors
Loading required package: stats4

Attaching package: 'S4Vectors'

The following objects are masked from 'package:lubridate':

    second, second&lt;-

The following objects are masked from 'package:dplyr':

    first, rename

The following object is masked from 'package:tidyr':

    expand

The following object is masked from 'package:utils':

    findMatches

The following objects are masked from 'package:base':

    expand.grid, I, unname

Loading required package: IRanges

Attaching package: 'IRanges'

The following object is masked from 'package:lubridate':

    %within%

The following objects are masked from 'package:dplyr':

    collapse, desc, slice

The following object is masked from 'package:purrr':

    reduce

The following object is masked from 'package:grDevices':

    windows</code></pre>
</div>
<div class="sourceCode cell-code" id="cb115"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb115-1"><a href="#cb115-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ERBS)</span>
<span id="cb115-2"><a href="#cb115-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(liftOver)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: GenomicRanges
Loading required package: rtracklayer</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'rtracklayer' was built under R version 4.3.1</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: Homo.sapiens
Loading required package: AnnotationDbi</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'AnnotationDbi' was built under R version 4.3.2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: Biobase
Welcome to Bioconductor

    Vignettes contain introductory material; view with
    'browseVignettes()'. To cite Bioconductor, see
    'citation("Biobase")', and for packages 'citation("pkgname")'.


Attaching package: 'AnnotationDbi'

The following object is masked from 'package:dplyr':

    select

Loading required package: OrganismDbi</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'OrganismDbi' was built under R version 4.3.1</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: GenomicFeatures</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'GenomicFeatures' was built under R version 4.3.2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: GO.db

Loading required package: org.Hs.eg.db

Loading required package: TxDb.Hsapiens.UCSC.hg19.knownGene</code></pre>
</div>
</div>
<p>First, we need to download the data, keep the 24 chromosomes (from 1 to Y) and, specify the sequence information from the GRCh38 human genome annotation.</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [13]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb125"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb125-1"><a href="#cb125-1" aria-hidden="true" tabindex="-1"></a>gwcat <span class="ot">=</span> <span class="fu">get_cached_gwascat</span>()</span>
<span id="cb125-2"><a href="#cb125-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-3"><a href="#cb125-3" aria-hidden="true" tabindex="-1"></a>gg <span class="ot">=</span> gwcat <span class="sc">|&gt;</span> <span class="fu">as_GRanges</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>dropping 45505 records that have NA for CHR_POS</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>1950 records have semicolon in CHR_POS; splitting and using first entry.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>3265 records have ' x ' in CHR_POS indicating multiple SNP effects, using first.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb129"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb129-1"><a href="#cb129-1" aria-hidden="true" tabindex="-1"></a>gg <span class="ot">=</span> <span class="fu">keepStandardChromosomes</span>(gg, <span class="at">pruning.mode =</span> <span class="st">"coarse"</span>)</span>
<span id="cb129-2"><a href="#cb129-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb129-3"><a href="#cb129-3" aria-hidden="true" tabindex="-1"></a><span class="co"># seqlevelsStyle(gg) &lt;- "UCSC"</span></span>
<span id="cb129-4"><a href="#cb129-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb129-5"><a href="#cb129-5" aria-hidden="true" tabindex="-1"></a><span class="fu">seqlevels</span>(gg) <span class="ot">&lt;-</span> <span class="fu">seqlevels</span>(gg) <span class="sc">|&gt;</span> </span>
<span id="cb129-6"><a href="#cb129-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sortSeqlevels</span>()</span>
<span id="cb129-7"><a href="#cb129-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb129-8"><a href="#cb129-8" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">"si.hs.38"</span>)</span>
<span id="cb129-9"><a href="#cb129-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb129-10"><a href="#cb129-10" aria-hidden="true" tabindex="-1"></a><span class="fu">seqinfo</span>(gg) <span class="ot">&lt;-</span> si.hs<span class="fl">.38</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<p>Now, let’s plot a karyogram that will show the SNP’s identified with significant associations with a phenotype. The SNP’s in the GWAS catalog have a stringent criterion of significance and there has been a replication of the finding from a independent population.</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [14]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb130"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb130-1"><a href="#cb130-1" aria-hidden="true" tabindex="-1"></a>ggbio<span class="sc">::</span><span class="fu">autoplot</span>(gg, <span class="at">layout=</span><span class="st">"karyogram"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Registered S3 method overwritten by 'GGally':
  method from   
  +.gg   ggplot2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Scale for x is already present.
Adding another scale for x, which will replace the existing scale.
Scale for x is already present.
Adding another scale for x, which will replace the existing scale.</code></pre>
</div>
<div class="cell-output-display">
<div class="">
<figure class="figure">
<p class=""><a href="index_files/figure-html/unnamed-chunk-3-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-2"><img src="index_files/figure-html/unnamed-chunk-3-1.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
</div></div>
<p>We can see the peak data as a <code>GRanges</code> object:</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [15]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb133"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb133-1"><a href="#cb133-1" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">"GM12878"</span>)</span>
<span id="cb133-2"><a href="#cb133-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb133-3"><a href="#cb133-3" aria-hidden="true" tabindex="-1"></a>GM12878</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>GRanges object with 1873 ranges and 7 metadata columns:
         seqnames            ranges strand |      name     score       col
            &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;integer&gt; &lt;logical&gt;
     [1]     chrX   1509355-1512462      * |         5         0      &lt;NA&gt;
     [2]     chrX 26801422-26802448      * |         6         0      &lt;NA&gt;
     [3]    chr19 11694102-11695359      * |         1         0      &lt;NA&gt;
     [4]    chr19   4076893-4079276      * |         4         0      &lt;NA&gt;
     [5]     chr3 53288568-53290767      * |         9         0      &lt;NA&gt;
     ...      ...               ...    ... .       ...       ...       ...
  [1869]    chr19 11201120-11203985      * |      8701         0      &lt;NA&gt;
  [1870]    chr19   2234920-2237370      * |       990         0      &lt;NA&gt;
  [1871]     chr1 94311336-94313543      * |      4035         0      &lt;NA&gt;
  [1872]    chr19 45690614-45691210      * |     10688         0      &lt;NA&gt;
  [1873]    chr19   6110100-6111252      * |      2274         0      &lt;NA&gt;
         signalValue    pValue    qValue      peak
           &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;integer&gt;
     [1]      157.92   310.000        32      1991
     [2]      147.38   310.000        32       387
     [3]       99.71   311.660        32       861
     [4]       84.74   310.000        32      1508
     [5]       78.20   299.505        32      1772
     ...         ...       ...       ...       ...
  [1869]        8.65     7.281   0.26576      2496
  [1870]        8.65    26.258   1.99568      1478
  [1871]        8.65    12.511   1.47237      1848
  [1872]        8.65     6.205   0.00000       298
  [1873]        8.65    17.356   2.01323       496
  -------
  seqinfo: 93 sequences (1 circular) from hg19 genome</code></pre>
</div>
</div></div>
<p>If we see the bottom of the <code>GRanges</code> table, this experiment have the hg19 annotation from the human genome. To work on the GRCh38 annotation we need to lift-over with a <code>.chain</code> file. For this we can use the <code>AnnotationHub</code> package.</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [16]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb135"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb135-1"><a href="#cb135-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(AnnotationHub)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'AnnotationHub' was built under R version 4.3.1</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: BiocFileCache</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'BiocFileCache' was built under R version 4.3.1</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: dbplyr</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'dbplyr' was built under R version 4.3.2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'dbplyr'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:dplyr':

    ident, sql</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'AnnotationHub'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:Biobase':

    cache</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:rtracklayer':

    hubUrl</code></pre>
</div>
<div class="sourceCode cell-code" id="cb146"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb146-1"><a href="#cb146-1" aria-hidden="true" tabindex="-1"></a>ah <span class="ot">&lt;-</span> AnnotationHub<span class="sc">::</span><span class="fu">AnnotationHub</span>()</span>
<span id="cb146-2"><a href="#cb146-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb146-3"><a href="#cb146-3" aria-hidden="true" tabindex="-1"></a><span class="fu">query</span>(ah, <span class="fu">c</span>(<span class="st">"hg19ToHg38.over.chain"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>AnnotationHub with 1 record
# snapshotDate(): 2023-10-23
# names(): AH14150
# $dataprovider: UCSC
# $species: Homo sapiens
# $rdataclass: ChainFile
# $rdatadateadded: 2014-12-15
# $title: hg19ToHg38.over.chain.gz
# $description: UCSC liftOver chain file from hg19 to hg38
# $taxonomyid: 9606
# $genome: hg19
# $sourcetype: Chain
# $sourceurl: http://hgdownload.cse.ucsc.edu/goldenpath/hg19/liftOver/hg19To...
# $sourcesize: NA
# $tags: c("liftOver", "chain", "UCSC", "genome", "homology") 
# retrieve record with 'object[["AH14150"]]' </code></pre>
</div>
<div class="sourceCode cell-code" id="cb148"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb148-1"><a href="#cb148-1" aria-hidden="true" tabindex="-1"></a>chain <span class="ot">&lt;-</span> ah[[<span class="st">"AH14150"</span>]]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>loading from cache</code></pre>
</div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [17]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb150"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb150-1"><a href="#cb150-1" aria-hidden="true" tabindex="-1"></a>GM12878 <span class="ot">&lt;-</span> <span class="fu">liftOver</span>(GM12878, chain) <span class="sc">|&gt;</span> </span>
<span id="cb150-2"><a href="#cb150-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unlist</span>()</span>
<span id="cb150-3"><a href="#cb150-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb150-4"><a href="#cb150-4" aria-hidden="true" tabindex="-1"></a><span class="fu">seqlevelsStyle</span>(GM12878) <span class="ot">&lt;-</span> <span class="st">"NCBI"</span></span>
<span id="cb150-5"><a href="#cb150-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb150-6"><a href="#cb150-6" aria-hidden="true" tabindex="-1"></a><span class="fu">seqinfo</span>(GM12878) <span class="ot">&lt;-</span> si.hs<span class="fl">.38</span></span>
<span id="cb150-7"><a href="#cb150-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb150-8"><a href="#cb150-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb150-9"><a href="#cb150-9" aria-hidden="true" tabindex="-1"></a><span class="fu">seqlevelsStyle</span>(GM12878) <span class="ot">&lt;-</span> <span class="st">"UCSC"</span></span>
<span id="cb150-10"><a href="#cb150-10" aria-hidden="true" tabindex="-1"></a><span class="fu">seqlevelsStyle</span>(gg) <span class="ot">&lt;-</span> <span class="st">"UCSC"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<p>We can find overlaps between the GWAS catalog and the ESRRA ChIP-Seq experiment but, there is a problem; the GWAS catalog is a collection of intervals that reports all significant SNPs and there can be duplications of SNPs associated to multiple phenotypes or the same SNP might be found for the same phenotype in different studies.</p>
<p>We can see the duplications with the <code>reduce</code> function from <code>IRanges</code> package:</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [18]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb151"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb151-1"><a href="#cb151-1" aria-hidden="true" tabindex="-1"></a><span class="co"># duplicated loci</span></span>
<span id="cb151-2"><a href="#cb151-2" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(gg) <span class="sc">-</span> <span class="fu">length</span>(<span class="fu">reduce</span>(gg))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 261160</code></pre>
</div>
</div></div>
<p>We can see that there are <code>261160</code> duplicated loci. Let’s find the overlap between the <em>reduced</em> catalog and the ChIP-Seq experiment:</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [19]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb153"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb153-1"><a href="#cb153-1" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb153-2"><a href="#cb153-2" aria-hidden="true" tabindex="-1"></a>fo <span class="ot">=</span> <span class="fu">findOverlaps</span>(GM12878, <span class="fu">reduce</span>(gg))</span>
<span id="cb153-3"><a href="#cb153-3" aria-hidden="true" tabindex="-1"></a>fo</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Hits object with 613 hits and 0 metadata columns:
        queryHits subjectHits
        &lt;integer&gt;   &lt;integer&gt;
    [1]         6      237757
    [2]         9      221130
    [3]        12       25060
    [4]        15      104699
    [5]        15      104700
    ...       ...         ...
  [609]      1928      246305
  [610]      1929      244698
  [611]      1929      244699
  [612]      1931      250380
  [613]      1931      250381
  -------
  queryLength: 1932 / subjectLength: 268560</code></pre>
</div>
</div></div>
<p>We can see 613 hits. Then, we are gonna eobtain the ranges from those hits, retrieve the phenotypes (DISEASE/TRAIT) and show the top 20 most common phenotypes with association to SNPs that lies on the ESRRA binding peaks.</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [20]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb155"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb155-1"><a href="#cb155-1" aria-hidden="true" tabindex="-1"></a>over_ranges <span class="ot">&lt;-</span> <span class="fu">reduce</span>(gg)[<span class="fu">subjectHits</span>(fo)]</span>
<span id="cb155-2"><a href="#cb155-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb155-3"><a href="#cb155-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb155-4"><a href="#cb155-4" aria-hidden="true" tabindex="-1"></a>ii <span class="ot">&lt;-</span> over_ranges <span class="sc">|&gt;</span> </span>
<span id="cb155-5"><a href="#cb155-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">|&gt;</span> </span>
<span id="cb155-6"><a href="#cb155-6" aria-hidden="true" tabindex="-1"></a>  GenomicRanges<span class="sc">::</span><span class="fu">makeGRangesListFromDataFrame</span>()</span>
<span id="cb155-7"><a href="#cb155-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb155-8"><a href="#cb155-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb155-9"><a href="#cb155-9" aria-hidden="true" tabindex="-1"></a>phset <span class="ot">=</span> <span class="fu">lapply</span>(ii, <span class="cf">function</span>(x){</span>
<span id="cb155-10"><a href="#cb155-10" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb155-11"><a href="#cb155-11" aria-hidden="true" tabindex="-1"></a>  <span class="co"># print(glue::glue("On range: {x}"))</span></span>
<span id="cb155-12"><a href="#cb155-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unique</span>(gg[<span class="fu">which</span>(gg <span class="sc">%over%</span> x)]<span class="sc">$</span><span class="st">"DISEASE/TRAIT"</span>)</span>
<span id="cb155-13"><a href="#cb155-13" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb155-14"><a href="#cb155-14" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb155-15"><a href="#cb155-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb155-16"><a href="#cb155-16" aria-hidden="true" tabindex="-1"></a>gwas_on_peaks <span class="ot">&lt;-</span> phset <span class="sc">|&gt;</span> </span>
<span id="cb155-17"><a href="#cb155-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">enframe</span>() <span class="sc">|&gt;</span> </span>
<span id="cb155-18"><a href="#cb155-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">unnest</span>(value) <span class="sc">|&gt;</span> </span>
<span id="cb155-19"><a href="#cb155-19" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">count</span>(value) <span class="sc">|&gt;</span> </span>
<span id="cb155-20"><a href="#cb155-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">slice_max</span>(n, <span class="at">n=</span><span class="dv">20</span>)</span>
<span id="cb155-21"><a href="#cb155-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb155-22"><a href="#cb155-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb155-23"><a href="#cb155-23" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> gwas_on_peaks <span class="sc">|&gt;</span> </span>
<span id="cb155-24"><a href="#cb155-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">value =</span> <span class="fu">fct_reorder</span>(value,n)) <span class="sc">|&gt;</span> </span>
<span id="cb155-25"><a href="#cb155-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(n,value, <span class="at">fill=</span>value))<span class="sc">+</span></span>
<span id="cb155-26"><a href="#cb155-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>() <span class="sc">+</span></span>
<span id="cb155-27"><a href="#cb155-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb155-28"><a href="#cb155-28" aria-hidden="true" tabindex="-1"></a>  paletteer<span class="sc">::</span><span class="fu">scale_fill_paletteer_d</span>(<span class="st">"khroma::smoothrainbow"</span>) <span class="sc">+</span></span>
<span id="cb155-29"><a href="#cb155-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span>
<span id="cb155-30"><a href="#cb155-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb155-31"><a href="#cb155-31" aria-hidden="true" tabindex="-1"></a>bslib<span class="sc">::</span><span class="fu">card</span>(plotly<span class="sc">::</span><span class="fu">ggplotly</span>(p), <span class="at">full_screen =</span> T)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="card bslib-card bslib-mb-spacing html-fill-item html-fill-container" data-bslib-card-init="" data-full-screen="false" data-require-bs-caller="card()" data-require-bs-version="5">
<div class="card-body bslib-gap-spacing html-fill-item html-fill-container" style="margin-top:auto;margin-bottom:auto;flex:1 1 auto;">
<div class="plotly html-widget html-fill-item" id="htmlwidget-171c168db4a533b58e5b" style="width:100%;height:400px;"></div>
<script type="application/json" data-for="htmlwidget-171c168db4a533b58e5b">{"x":{"data":[{"orientation":"v","width":7,"base":0.55000000000000004,"x":[3.5],"y":[0.89999999999999991],"text":"n:  7<br />value: Appendicular lean mass<br />value: Appendicular lean mass","type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(232,236,251,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"Appendicular lean mass","legendgroup":"Appendicular lean mass","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":7,"base":1.55,"x":[3.5],"y":[0.90000000000000013],"text":"n:  7<br />value: Diastolic blood pressure<br />value: Diastolic blood pressure","type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(221,216,239,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"Diastolic blood pressure","legendgroup":"Diastolic blood pressure","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":7,"base":2.5499999999999998,"x":[3.5],"y":[0.90000000000000036],"text":"n:  7<br />value: Educational attainment<br />value: Educational attainment","type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(209,193,225,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"Educational attainment","legendgroup":"Educational attainment","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":7,"base":3.5499999999999998,"x":[3.5],"y":[0.90000000000000036],"text":"n:  7<br />value: Lymphocyte count<br />value: Lymphocyte count","type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(195,168,209,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"Lymphocyte count","legendgroup":"Lymphocyte count","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":7,"base":4.5499999999999998,"x":[3.5],"y":[0.90000000000000036],"text":"n:  7<br />value: Mean spheric corpuscular volume<br />value: Mean spheric corpuscular volume","type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(181,143,194,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"Mean spheric corpuscular volume","legendgroup":"Mean spheric corpuscular volume","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":7,"base":5.5499999999999998,"x":[3.5],"y":[0.90000000000000036],"text":"n:  7<br />value: Monocyte count<br />value: Monocyte count","type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(167,120,180,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"Monocyte count","legendgroup":"Monocyte count","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":7,"base":6.5499999999999998,"x":[3.5],"y":[0.90000000000000036],"text":"n:  7<br />value: Plateletcrit<br />value: Plateletcrit","type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(155,98,167,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"Plateletcrit","legendgroup":"Plateletcrit","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":7,"base":7.5499999999999998,"x":[3.5],"y":[0.89999999999999947],"text":"n:  7<br />value: Reticulocyte fraction of red cells<br />value: Reticulocyte fraction of red cells","type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(140,78,153,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"Reticulocyte fraction of red cells","legendgroup":"Reticulocyte fraction of red cells","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":7,"base":8.5500000000000007,"x":[3.5],"y":[0.89999999999999858],"text":"n:  7<br />value: Rheumatoid arthritis<br />value: Rheumatoid arthritis","type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(111,76,155,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"Rheumatoid arthritis","legendgroup":"Rheumatoid arthritis","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":7,"base":9.5500000000000007,"x":[3.5],"y":[0.89999999999999858],"text":"n:  7<br />value: Triglyceride levels<br />value: Triglyceride levels","type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(96,89,169,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"Triglyceride levels","legendgroup":"Triglyceride levels","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":7,"base":10.550000000000001,"x":[3.5],"y":[0.89999999999999858],"text":"n:  7<br />value: Type 2 diabetes<br />value: Type 2 diabetes","type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(85,104,184,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"Type 2 diabetes","legendgroup":"Type 2 diabetes","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":8,"base":11.550000000000001,"x":[4],"y":[0.89999999999999858],"text":"n:  8<br />value: Heel bone mineral density<br />value: Heel bone mineral density","type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(78,121,197,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"Heel bone mineral density","legendgroup":"Heel bone mineral density","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":8,"base":12.550000000000001,"x":[4],"y":[0.89999999999999858],"text":"n:  8<br />value: Systolic blood pressure<br />value: Systolic blood pressure","type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(77,138,198,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"Systolic blood pressure","legendgroup":"Systolic blood pressure","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":9,"base":13.550000000000001,"x":[4.5],"y":[0.89999999999999858],"text":"n:  9<br />value: Sex hormone-binding globulin levels<br />value: Sex hormone-binding globulin levels","type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(78,150,188,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"Sex hormone-binding globulin levels","legendgroup":"Sex hormone-binding globulin levels","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":10,"base":14.550000000000001,"x":[5],"y":[0.89999999999999858],"text":"n: 10<br />value: Eosinophil counts<br />value: Eosinophil counts","type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(84,158,179,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"Eosinophil counts","legendgroup":"Eosinophil counts","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":10,"base":15.550000000000001,"x":[5],"y":[0.89999999999999858],"text":"n: 10<br />value: HDL cholesterol levels<br />value: HDL cholesterol levels","type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(89,165,169,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"HDL cholesterol levels","legendgroup":"HDL cholesterol levels","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":10,"base":16.550000000000001,"x":[5],"y":[0.89999999999999858],"text":"n: 10<br />value: Hematocrit<br />value: Hematocrit","type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(96,171,158,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"Hematocrit","legendgroup":"Hematocrit","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":10,"base":17.550000000000001,"x":[5],"y":[0.89999999999999858],"text":"n: 10<br />value: Platelet count<br />value: Platelet count","type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(105,177,144,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"Platelet count","legendgroup":"Platelet count","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":10,"base":18.550000000000001,"x":[5],"y":[0.89999999999999858],"text":"n: 10<br />value: Total cholesterol levels<br />value: Total cholesterol levels","type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(119,183,125,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"Total cholesterol levels","legendgroup":"Total cholesterol levels","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":10,"base":19.550000000000001,"x":[5],"y":[0.89999999999999858],"text":"n: 10<br />value: Waist-to-hip ratio adjusted for BMI<br />value: Waist-to-hip ratio adjusted for BMI","type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(140,188,104,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"Waist-to-hip ratio adjusted for BMI","legendgroup":"Waist-to-hip ratio adjusted for BMI","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":10,"base":20.550000000000001,"x":[5],"y":[0.89999999999999858],"text":"n: 10<br />value: White blood cell count<br />value: White blood cell count","type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(166,190,84,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"White blood cell count","legendgroup":"White blood cell count","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":11,"base":21.550000000000001,"x":[5.5],"y":[0.89999999999999858],"text":"n: 11<br />value: High density lipoprotein cholesterol levels<br />value: High density lipoprotein cholesterol levels","type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(190,188,72,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"High density lipoprotein cholesterol levels","legendgroup":"High density lipoprotein cholesterol levels","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":13,"base":22.550000000000001,"x":[6.5],"y":[0.89999999999999858],"text":"n: 13<br />value: Multi-trait sex score<br />value: Multi-trait sex score","type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(209,181,65,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"Multi-trait sex score","legendgroup":"Multi-trait sex score","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":14,"base":23.550000000000001,"x":[7],"y":[0.89999999999999858],"text":"n: 14<br />value: Body mass index<br />value: Body mass index","type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(221,170,60,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"Body mass index","legendgroup":"Body mass index","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":15,"base":24.550000000000001,"x":[7.5],"y":[0.89999999999999858],"text":"n: 15<br />value: Red cell distribution width<br />value: Red cell distribution width","type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(228,156,57,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"Red cell distribution width","legendgroup":"Red cell distribution width","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":16,"base":25.550000000000001,"x":[8],"y":[0.89999999999999858],"text":"n: 16<br />value: Mean corpuscular volume<br />value: Mean corpuscular volume","type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(231,140,53,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"Mean corpuscular volume","legendgroup":"Mean corpuscular volume","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":18,"base":26.550000000000001,"x":[9],"y":[0.89999999999999858],"text":"n: 18<br />value: Red blood cell count<br />value: Red blood cell count","type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(230,121,50,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"Red blood cell count","legendgroup":"Red blood cell count","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":25,"base":27.550000000000001,"x":[12.5],"y":[0.89999999999999858],"text":"n: 25<br />value: Mean corpuscular hemoglobin<br />value: Mean corpuscular hemoglobin","type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(228,99,45,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"Mean corpuscular hemoglobin","legendgroup":"Mean corpuscular hemoglobin","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":48,"base":28.550000000000001,"x":[24],"y":[0.89999999999999858],"text":"n: 48<br />value: Height<br />value: Height","type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(223,72,40,1)","line":{"width":1.8897637795275593,"color":"transparent"}},"name":"Height","legendgroup":"Height","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null}],"layout":{"margin":{"t":26.228310502283104,"r":7.3059360730593621,"b":40.182648401826491,"l":276.89497716894988},"font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724},"xaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[-2.4000000000000004,50.399999999999999],"tickmode":"array","ticktext":["0","10","20","30","40","50"],"tickvals":[0,10,20,30,40,50],"categoryorder":"array","categoryarray":["0","10","20","30","40","50"],"nticks":null,"ticks":"","tickcolor":null,"ticklen":3.6529680365296811,"tickwidth":0,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.68949771689498},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":true,"gridcolor":"rgba(235,235,235,1)","gridwidth":0.66417600664176002,"zeroline":false,"anchor":"y","title":{"text":"n","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724}},"hoverformat":".2f"},"yaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[0.40000000000000002,29.600000000000001],"tickmode":"array","ticktext":["Appendicular lean mass","Diastolic blood pressure","Educational attainment","Lymphocyte count","Mean spheric corpuscular volume","Monocyte count","Plateletcrit","Reticulocyte fraction of red cells","Rheumatoid arthritis","Triglyceride levels","Type 2 diabetes","Heel bone mineral density","Systolic blood pressure","Sex hormone-binding globulin levels","Eosinophil counts","HDL cholesterol levels","Hematocrit","Platelet count","Total cholesterol levels","Waist-to-hip ratio adjusted for BMI","White blood cell count","High density lipoprotein cholesterol levels","Multi-trait sex score","Body mass index","Red cell distribution width","Mean corpuscular volume","Red blood cell count","Mean corpuscular hemoglobin","Height"],"tickvals":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29],"categoryorder":"array","categoryarray":["Appendicular lean mass","Diastolic blood pressure","Educational attainment","Lymphocyte count","Mean spheric corpuscular volume","Monocyte count","Plateletcrit","Reticulocyte fraction of red cells","Rheumatoid arthritis","Triglyceride levels","Type 2 diabetes","Heel bone mineral density","Systolic blood pressure","Sex hormone-binding globulin levels","Eosinophil counts","HDL cholesterol levels","Hematocrit","Platelet count","Total cholesterol levels","Waist-to-hip ratio adjusted for BMI","White blood cell count","High density lipoprotein cholesterol levels","Multi-trait sex score","Body mass index","Red cell distribution width","Mean corpuscular volume","Red blood cell count","Mean corpuscular hemoglobin","Height"],"nticks":null,"ticks":"","tickcolor":null,"ticklen":3.6529680365296811,"tickwidth":0,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.68949771689498},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":true,"gridcolor":"rgba(235,235,235,1)","gridwidth":0.66417600664176002,"zeroline":false,"anchor":"x","title":{"text":"value","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724}},"hoverformat":".2f"},"shapes":[{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":1}],"showlegend":true,"legend":{"bgcolor":null,"bordercolor":null,"borderwidth":0,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.68949771689498},"title":{"text":"value","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724}}},"hovermode":"closest","barmode":"relative"},"config":{"doubleClick":"reset","modeBarButtonsToAdd":["hoverclosest","hovercompare"],"showSendToCloud":false},"source":"A","attrs":{"4a54687b65e1":{"x":{},"y":{},"fill":{},"type":"bar"}},"cur_data":"4a54687b65e1","visdat":{"4a54687b65e1":["function (y) ","x"]},"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.20000000000000001,"selected":{"opacity":1},"debounce":0},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
</div>
<bslib-tooltip placement="auto" bsoptions="[]" data-require-bs-version="5" data-require-bs-caller="tooltip()">
<template>Expand</template>
<span class="bslib-full-screen-enter badge rounded-pill"><svg xmlns="http://www.w3.org/2000/svg" viewbox="0 0 24 24" style="height:1em;width:1em;fill:currentColor;" aria-hidden="true" role="img"><path d="M20 5C20 4.4 19.6 4 19 4H13C12.4 4 12 3.6 12 3C12 2.4 12.4 2 13 2H21C21.6 2 22 2.4 22 3V11C22 11.6 21.6 12 21 12C20.4 12 20 11.6 20 11V5ZM4 19C4 19.6 4.4 20 5 20H11C11.6 20 12 20.4 12 21C12 21.6 11.6 22 11 22H3C2.4 22 2 21.6 2 21V13C2 12.4 2.4 12 3 12C3.6 12 4 12.4 4 13V19Z"></path></svg></span>
</bslib-tooltip>
<script data-bslib-card-init="">bslib.Card.initializeAllCards();</script>
</div>
</div>
</div></div>
<p>Distinct phenotypes identified on the peaks:</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [21]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb156"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb156-1"><a href="#cb156-1" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(phset)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 613</code></pre>
</div>
</div></div>
<p>Now, how to do the inference of these phenotype on peaks of these b cells? We can use permutation on the genomic positions to test if the number of phenotypes found is due to chance or not.</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [22]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb158"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb158-1"><a href="#cb158-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ph525x)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: png</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: grid</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: VariantAnnotation</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'VariantAnnotation' was built under R version 4.3.2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: MatrixGenerics</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'MatrixGenerics' was built under R version 4.3.1</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: matrixStats</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'matrixStats' was built under R version 4.3.2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'matrixStats'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:Biobase':

    anyMissing, rowMedians</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:dplyr':

    count</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'MatrixGenerics'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:matrixStats':

    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,
    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,
    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,
    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,
    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,
    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,
    colWeightedMeans, colWeightedMedians, colWeightedSds,
    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,
    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,
    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,
    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,
    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,
    rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,
    rowWeightedMads, rowWeightedMeans, rowWeightedMedians,
    rowWeightedSds, rowWeightedVars</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:Biobase':

    rowMedians</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: SummarizedExperiment</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'SummarizedExperiment' was built under R version 4.3.1</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: Rsamtools</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'Rsamtools' was built under R version 4.3.1</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: Biostrings</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'Biostrings' was built under R version 4.3.2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: XVector</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'XVector' was built under R version 4.3.1</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'XVector'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:purrr':

    compact</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'Biostrings'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:grid':

    pattern</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:base':

    strsplit</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'VariantAnnotation'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:stringr':

    fixed</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:base':

    tabulate</code></pre>
</div>
<div class="sourceCode cell-code" id="cb189"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb189-1"><a href="#cb189-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(progress)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'progress' was built under R version 4.3.3</code></pre>
</div>
<div class="sourceCode cell-code" id="cb191"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb191-1"><a href="#cb191-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb191-2"><a href="#cb191-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb191-3"><a href="#cb191-3" aria-hidden="true" tabindex="-1"></a>n_iter <span class="ot">=</span> <span class="dv">100</span></span>
<span id="cb191-4"><a href="#cb191-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb191-5"><a href="#cb191-5" aria-hidden="true" tabindex="-1"></a>pb <span class="ot">&lt;-</span> progress_bar<span class="sc">$</span><span class="fu">new</span>(<span class="at">format =</span> <span class="st">"(:spin) [:bar] :percent [Elapsed time: :elapsedfull || Estimated time remaining: :eta]"</span>,</span>
<span id="cb191-6"><a href="#cb191-6" aria-hidden="true" tabindex="-1"></a>                       <span class="at">total =</span> n_iter,</span>
<span id="cb191-7"><a href="#cb191-7" aria-hidden="true" tabindex="-1"></a>                       <span class="at">complete =</span> <span class="st">"="</span>,   <span class="co"># Completion bar character</span></span>
<span id="cb191-8"><a href="#cb191-8" aria-hidden="true" tabindex="-1"></a>                       <span class="at">incomplete =</span> <span class="st">" "</span>, <span class="co"># Incomplete bar character</span></span>
<span id="cb191-9"><a href="#cb191-9" aria-hidden="true" tabindex="-1"></a>                       <span class="at">current =</span> <span class="st">"&gt;"</span>,    <span class="co"># Current bar character</span></span>
<span id="cb191-10"><a href="#cb191-10" aria-hidden="true" tabindex="-1"></a>                       <span class="at">clear =</span> <span class="cn">FALSE</span>,    <span class="co"># If TRUE, clears the bar when finish</span></span>
<span id="cb191-11"><a href="#cb191-11" aria-hidden="true" tabindex="-1"></a>                       <span class="at">width =</span> <span class="dv">100</span>)   </span>
<span id="cb191-12"><a href="#cb191-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb191-13"><a href="#cb191-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb191-14"><a href="#cb191-14" aria-hidden="true" tabindex="-1"></a>rsc <span class="ot">=</span> <span class="fu">sapply</span>(<span class="dv">1</span><span class="sc">:</span>n_iter, <span class="cf">function</span>(x) {</span>
<span id="cb191-15"><a href="#cb191-15" aria-hidden="true" tabindex="-1"></a>    pb<span class="sc">$</span><span class="fu">tick</span>()</span>
<span id="cb191-16"><a href="#cb191-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">length</span>(<span class="fu">findOverlaps</span>(<span class="fu">reposition</span>(GM12878), <span class="fu">reduce</span>(gg))) <span class="sc">|&gt;</span> </span>
<span id="cb191-17"><a href="#cb191-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">suppressWarnings</span>()</span>
<span id="cb191-18"><a href="#cb191-18" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb191-19"><a href="#cb191-19" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb191-20"><a href="#cb191-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb191-21"><a href="#cb191-21" aria-hidden="true" tabindex="-1"></a><span class="co"># compute prop with more overlaps than in observed data</span></span>
<span id="cb191-22"><a href="#cb191-22" aria-hidden="true" tabindex="-1"></a><span class="fu">mean</span>(rsc <span class="sc">&gt;</span> <span class="fu">length</span>(fo))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0</code></pre>
</div>
</div></div>
<section id="explore-the-tcga-1" class="level2" data-number="4">
<h2 data-number="4" class="anchored" data-anchor-id="explore-the-tcga-1"><span class="header-section-number">4</span> Explore the TCGA</h2>
<p>Please check the script below the image to see how to explore and get insights from the TCGA (The Cancer Genome Atlas) database.</p>
<p>, <code>[</code>, <code>[[</code> - extract colData columns, subset, or experiment *Format() - convert into a long or wide DataFrame assays() - convert ExperimentList to a SimpleList of matrices exportClass() - save data to flat files ```</p>
</section>
</div></div>
</div>
<p>:::</p>
</section>
<section id="conclusion" class="level2" data-number="5">
<h2 data-number="5" class="anchored" data-anchor-id="conclusion"><span class="header-section-number">5</span> Conclusion</h2>
</section>
<section id="references" class="level2 unnumbered">
<h2 class="unnumbered anchored" data-anchor-id="references">References</h2>
<div id="refs" role="list">

</div>
</section>
     </main>
<!-- /main column -->  <script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>  </div> <!-- /content -->  <script>var lightboxQuarto = GLightbox({"openEffect":"zoom","selector":".lightbox","loop":false,"closeEffect":"zoom","descPosition":"bottom"});
window.onload = () => {
  lightboxQuarto.on('slide_before_load', (data) => {
    const { slideIndex, slideNode, slideConfig, player, trigger } = data;
    const href = trigger.getAttribute('href');
    if (href !== null) {
      const imgEl = window.document.querySelector(`a[href="${href}"] img`);
      if (imgEl !== null) {
        const srcAttr = imgEl.getAttribute("src");
        if (srcAttr && srcAttr.startsWith("data:")) {
          slideConfig.href = srcAttr;
        }
      }
    } 
  });

  lightboxQuarto.on('slide_after_load', (data) => {
    const { slideIndex, slideNode, slideConfig, player, trigger } = data;
    if (window.Quarto?.typesetMath) {
      window.Quarto.typesetMath(slideNode);
    }
  });

};
          </script> 
  
</body></html>